docs/assemblies/assembly-prometheus-metrics-proxy.adoc

```diff
@@ -0,0 +1,88 @@
+// file included in the following:
+//
+// kroxylicious-operator/index.adoc
+
+[id='assembly-prometheus-metrics-proxy-{context}']
+= Proxy Metrics
+
+[role="_abstract"]
+
+The proxy provides *connections metrics* and *message metrics*.
+
+Connection metrics allow the connections made from the clients to the proxy (the downstream) and the connection made by the proxy to the kafka cluster (the upstream) to be understood.
+
+Message metrics allow the {kafka-protocol}#protocol_messages[Kafka protocol messages] that flow through the proxy to be understood.
+
+== Connection Metrics
+
+The connection metrics count the connections made to the proxy by the clients, and by the proxy to the brokers.
+
+If an error occurs whilst the accepting the connection from the client, the client error counter will tick.
+If an error occurs whilst forming the connection to the brokers, the server error counter will tick.
+In the case of an error, the corresponding connection counter will tick too.
+
+Connection metrics are discriminated by the virtual cluster's name (`virtual_cluster`) and the broker's node ID (`node_id`).
+When the client connects to boostrap endpoint of the virtual cluster, a node ID value of `bootstrap` is recorded.
+
+On the client end, if an error occurs before the virtual cluster has been established, the error counter will tick
+with the virtual cluster and node ID labels set to the empty string`""`.
+
+|===
+|Metric Name |Type |Labels|Description
+
+|`kroxylicious_client_to_proxy_connection_total`
+|Counter
+|`virtual_cluster`, `node_id`
+|Incremented by one every time a connection is accepted from a downstream client by the proxy.
+
+|`kroxylicious_client_to_proxy_errors_total`
+|Counter
+|`virtual_cluster`, `node_id`
+|Incremented by one every time a connection is closed due to any downstream error.
+
+|`kroxylicious_proxy_to_server_connections_total`
+|Counter
+|`virtual_cluster`, `node_id`
+|Incremented by one every time a connection is made upstream to the server from the proxy.
```

Comment:
Should we mention that metrics also count early fails? and also I guess maybe mentioning something like `Incremented by one every time a connection (accept/attempt/early fails) is made upstream to the server from the proxy.`